pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -159,13 +172,42 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 				WithValueFrom(corev1ac.EnvVarSource().
 					WithFieldRef(corev1ac.ObjectFieldSelector().
 						WithFieldPath(constants.JobCompletionIndexFieldPath))),
-			*corev1ac.EnvVar().
-				WithName(constants.TorchEnvMasterAddr).
-				WithValue(fmt.Sprintf("%s-%s-0-0.%s", trainJob.Name, constants.Node, trainJob.Name)),
-			*corev1ac.EnvVar().
-				WithName(constants.TorchEnvMasterPort).
-				WithValue(fmt.Sprintf("%d", constants.ContainerTrainerPort)),
 		)
+
+		if !slices.Equal(trainJob.Spec.Trainer.Command, constants.TorchTuneEntrypoint) {
+			// Add PET_MASTER_ADDR and PET_MASTER_PORT envs for torchrun.
+			apply.UpsertEnvVar(&trainerContainer.Env,
+				*corev1ac.EnvVar().
+					WithName(constants.TorchEnvMasterAddr).
+					WithValue(fmt.Sprintf("%s-%s-0-0.%s", trainJob.Name, constants.Node, trainJob.Name)),
+				*corev1ac.EnvVar().
+					WithName(constants.TorchEnvMasterPort).
+					WithValue(fmt.Sprintf("%d", constants.ContainerTrainerPort)),
+			)
+		} else {
+			// Mutate command line args for torchtune.
+			// Ref: https://github.com/kubeflow/trainer/tree/master/docs/proposals/2401-llm-trainer-v2#complement-torch-plugin
+			oldArgs, newArgs := trainJob.Spec.Trainer.Args, []string{}
+
+			// 1. Add rendezvous backend arg for torchtune.
+			newArgs = append(newArgs,
+				fmt.Sprintf("%s %s-%s-0-0.%s:%d",
+					constants.TorchTuneArgRdzvEndpoint,
+					trainJob.Name, constants.Node, trainJob.Name, constants.ContainerTrainerPort,
+				),
+			)
+
+			// 2. Get the recipe and config from old args and append them to new args.
+			numNodes := ptr.Deref(ptr.Deref(trainerPS, runtime.PodSet{}).Count, 1)
+			recipe := getRecipeFromArgs(numNodes, numProcPerNode, oldArgs)
+			config := getConfigFileFromArgs(numNodes, recipe, trainJob.Spec.RuntimeRef.Name)
+			newArgs = append(newArgs, recipe, fmt.Sprintf("--config %s", config))
+
+			// 3. Reserve old arguments to override corresponding items in the config file.
+			newArgs = append(newArgs, oldArgs...)
```

Comment:
Would it be possible to assign newArgs to the container comand rather than args ? 
In that case, you don't need to perform this append.
e.g.
```yaml
container:
  command:
    - tune
    - run
    - lora_finetune_single_device
    - --config
    - llama3_2/3B_lora_single_device
args:
  - batch_size=8
```

Reply:
Yeah, but will it be simpler since you just need to append config + recipe to the command, and keep args as user sets ? 
Rather than insert oldArgs after the newArgs : https://github.com/kubeflow/trainer/blob/ccbbefbad992beb7b97cf2f9cb8766100fefd95c/pkg/runtime/framework/plugins/torch/torch.go#L190